homeassistant/components/homematicip_cloud/lock.py

```diff
@@ -0,0 +1,111 @@
+"""Support for HomematicIP Cloud lock devices."""
+from __future__ import annotations
+
+from typing import Any
+
+from homematicip.aio.device import AsyncDoorLockDrive
+from homematicip.base.enums import LockState, MotorState
+
+from homeassistant.components.lock import LockEntity, LockEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import DOMAIN as HMIPC_DOMAIN, HomematicipGenericEntity
+from .hap import HomematicipHAP
+
+ATTR_AUTO_RELOCK_DELAY = "auto_relock_delay"
+ATTR_DOOR_HANDLE_TYPE = "door_handle_type"
+ATTR_DOOR_LOCK_DIRECTION = "door_lock_direction"
+ATTR_DOOR_LOCK_NEUTRAL_POSITION = "door_lock_neutral_position"
+ATTR_DOOR_LOCK_TURNS = "door_lock_turns"
+
+DEVICE_DLD_ATTRIBUTES = {
+    "autoRelockDelay": ATTR_AUTO_RELOCK_DELAY,
+    "doorHandleType": ATTR_DOOR_HANDLE_TYPE,
+    "doorLockDirection": ATTR_DOOR_LOCK_DIRECTION,
+    "doorLockNeutralPosition": ATTR_DOOR_LOCK_NEUTRAL_POSITION,
+    "doorLockTurns": ATTR_DOOR_LOCK_TURNS,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the HomematicIP locks from a config entry."""
+    hap = hass.data[HMIPC_DOMAIN][config_entry.unique_id]
+    entities: list[HomematicipGenericEntity] = []
+    for device in hap.home.devices:
+        if isinstance(device, AsyncDoorLockDrive):
+            entities.append(HomematicipDoorLockDrive(hap, device))
+
+    if entities:
+        async_add_entities(entities)
+
+
+class HomematicipDoorLockDrive(HomematicipGenericEntity, LockEntity):
+    """Representation of the HomematicIP DoorLockDrive."""
+
+    _attr_supported_features = LockEntityFeature.OPEN
+
+    def __init__(
+        self,
+        hap: HomematicipHAP,
+        device,
+        post: str | None = None,
+        channel: int | None = None,
+        is_multi_channel: bool | None = False,
+    ) -> None:
+        """Initialize DoorLockDrive."""
+        super().__init__(hap, device)
```

Comment:
What is the purpose of overloading `__init__` here? You're not doing anything except init'ing the parent class.

Reply:
I'll remove that 👍